[docs] Improve docs about WSL and desktop app. (#3924)
organicmaps/organicmaps: docs/INSTALL.md

### Preparing

> You need a Linux or a Mac machine to build a desktop version of Organic Maps. Windows users can use the WSL.

Author: ok

You need a Linux or a Mac machine to build a desktop version of Organic Maps. Windows users can use the WSL.

> - We haven't compiled Organic Maps on Windows in a long time, though it is possible.

Author: This lines were there previously. I just moved them. 

I will change them to the suggestions you provided in later comments.

> Ensure that you have at least 20GB of free space.

Author: It is ;-) 
I'll write that in WSL you can follow steps in "Linux or Mac" section.

brew install cmake ninja qt@5
```

### Windows

We haven't compiled Organic Maps on Windows in a long time, though it is possible.
It is likely some make files should be updated.
If you succeed, please submit a tutorial.

> You'll need to have python3, cmake, ninja in the PATH and also to have Qt5 installed.

Reviewer: ```suggestion
You'll need to have python3, cmake, ninja, and QT5 in the PATH, and Visual Studio 2022 or Visual Studio 2022 Build Tools installed. Use [Visual Studio Developer Command Prompt](https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022) or generate Visual Studio project files with CMake to build the project.
```

However, it is possible to use the WSL to run GUI applications. 

> #### Windows 11

Reviewer: ```suggestion
#### Windows 11 (WSL)
```

Since Windows 11 Build 22000 running GUI apps from the WSL is supported [out-of-the box](https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps). After [building](#building-1) the app you should be able to [run](#running) it without any additional steps.

> #### Windows 10

Reviewer: ```suggestion
#### Windows 10 (WSL)
```

For Windows 10 you should do these steps (taken from [here](https://techcommunity.microsoft.com/t5/windows-dev-appconsult/running-wsl-gui-apps-on-windows-10/ba-p/1493242), check this blog post if you have any problems):
1. Download and install [VcXsrv Windows X Server](https://sourceforge.net/projects/vcxsrv/).
2. Run _XLaunch_ app to launch X Server. During settings make sure "Disable access control" checkbox is selected.
3. (optionally) Click "Save configuration" and save configuration to some file (for example to _config.xlaunch_). With this you will be able to quickly run desktop app in the future.
4. When asked about firewall, allow access for public and private networks.
5. Add this line:
    ```bash
    export DISPLAY=$(ip route|awk '/^default/{print $3}'):0.0
    ```
    to _/etc/bash.bashrc_ file.
6. Close WSL and enter it again.

Now when you want to run the desktop app you just need to first launch the X Server on Windows (for example, by running previously saved _config.xlaunch_ file) and then you should be able to [build](#building-1) and [run](#running) the app from WSL.

> Running X Server is also necessary to run `generate_symbols.sh` required when you change icons for [styles](STYLES.md)

Reviewer: ```suggestion
Running X Server is also required to run `generate_symbols.sh` script when you change icons for [styles](STYLES.md)
```

Some tests [are known to be broken](https://github.com/organicmaps/organicmaps/issues?q=is%3Aissue+is%3Aopen+label%3ATests).

> ### Changing themes

Reviewer: Should it be in contributing docs? Maybe add a link from here to there?

Author: This INSTALL.md file is the main place that describes desktop app so I think this "hidden" options should be documented here.

But I will change this to text to focus more on that these theme command are not only for theme changing
